### ISSUE TITLE
Remove as many `RelationConcept`s from the instance models as possible, by using `ModelKind` more

### ISSUE THREAD

**Issue by JacquesCarette** [JacquesCarette/Drasil#2371]
See https://github.com/JacquesCarette/Drasil/commit/f23bf7b5757b6686d4fa1a6aec371520300ee77b for example. This is part of adding `ModelKind` to `InstanceModel`. 

Basically, every instance model in all examples should be looked at, and either
1. its model should be changed from `OthModel` to something else (one per PR!)
2. an issue should be created regarding why it can't be done.

**Comment by balacij** [comment]
Trying to better keep track of them and their similarities:

Things not ticked off are still `OthModel`s.

- [ ] DblPendulum
    - [ ] GenDefs
        - [x] velocityIXGD - EquationalModel
        - [x] velocityIYGD - EquationalModel
        - [x] accelerationIXGD - EquationalModel
        - [x] accelerationIYGD - EquationalModel
        - [ ] hForceOnPendulumGD
        - [ ] vForceOnPendulumGD
        - [ ] angFrequencyGD
        - [ ] periodPend
   - [ ] IMods
       - [ ] angularDisplacementIM 
- [ ] GamePhysics
    - [x] GenDefs
        - [x] accelGravityGD - EquationalModel
        - [x] impulseGD  - EquationalModel
    - [ ] IMods
        - [ ] transMot
        - [ ] rotMot
        - [ ] col2D
    - [ ] TMods
        - [ ] newtonSL
        - [x] newtonTL - EquationalModel
        - [ ] newtonLUG
        - [ ] newtonSLR 
- [ ] GlassBR
    - [ ] IMods
        - [ ] pbIsSafe
        - [ ] lrIsSafe
    - [x] TMods
        - [x] pbIsSafe - EquationalModel
        - [x] lrIsSafe - EquationalModel
- [x] HGHC
- [ ] NoPCM
    - [ ] GenDefs
        - [ ] rocTempSimp
        - [ ] htFluxWaterFromCoil
    - [ ] IMods
        - [ ] eBalanceOnWtr
        - [ ] heatEInWtr
- [ ] PDController
    - [ ] GenDefs
        - [ ] gdPowerPlant
    - [ ] IMods
        - [ ] imPD
    - [ ] TMods
        - [ ] tmLaplace
        - [ ] tmInvLaplace
        - [ ] tmSOSystem
- [ ] Projectile
    - [ ] GenDefs
        - [ ] rectVelGD
        - [ ] rectPosGD
        - [ ] velVecGD
        - [ ] posVecGD
    - [ ] IMods
        - [ ] timeIM
        - [ ] landPosIM
        - [ ] offsetIM
        - [ ] messageIM
- [ ] SSP
    - [ ] GenDefs
        - [ ] normForcEqGD
        - [ ] bsShrFEqGD
        - [ ] resShrGD
        - [ ] mobShrGD
        - [ ] effNormFGD
        - [ ] resShearWOGD
        - [ ] mobShearWOGD
        - [ ] normShrRGD
        - [ ] momentEqlGD
        - [ ] weightGD
        - [ ] sliceWghtGD
        - [ ] hsPressureGD
        - [ ] baseWtrFGD
        - [ ] srfWtrFGD
    - [ ] IMods
        - [ ] fctSfty
        - [ ] nrmShrFor
        - [ ] nrmShrForNum
        - [ ] nrmShrForDen
        - [ ] intsliceFs
        - [ ] crtSlpId
    - [ ] TMods
        - [x] factOfSafety - EquationalModel
        - [ ] equilibrium
        - [ ] mcShrStrgth
        - [ ] effStress
        - [ ] newtonSL 
- [ ] SWHS
    - [ ] GenDefs
        - [ ] rocTempSimp
        - [ ] htFluxWaterFromCoil
        - [ ] htFluxPCMFromWater
    - [ ] IMods
        - [ ] eBalanceOnWtr
        - [ ] eBalanceOnPCM
        - [ ] heatEInWtr
        - [ ] heatEInPCM
    - [ ] TMods
        - [ ] consThermE
        - [ ] sensHtE
        - [ ] latentHtE
        - [ ] nwtnCooling

**Comment by balacij** [comment]
Note to self: Once all OthModels are converted into other things, we can easily implement MayHaveUnit of GenDefns in terms of the ModelKinds instead of having a separate field for it.

**Comment by balacij** [comment]
I don't remember where you (@JacquesCarette) mentioned it, but you previously mentioned that ModelKinds may need to be converted into a triple with a custom UID, NP (term), and the original ModelKind, and I think that with the last bunch of models, we will need that. Most of the existing converted ones only needed either a custom UID or term, so we added them to the models (IM/GD/TM) instead of the MK, but some of the last ones appear to need both.

I can see us changing `ModelsKinds` to something similar to:

```
type ModelKinds' = (ModelKinds, UID, NP)
```

or:

```
data ModelKinds' = MK' {
  _mk    :: ModelKinds,
  _mkUid :: UID,
  _mkNp  :: NP
}
```

The second variant is probably better.

**Comment by JacquesCarette** [comment]
Yes. As I've mentioned elsewhere, 4-tuples (and above): never. 3-tuples? rarely, and need strong justification.
